torchtext/experimental/functional.py

```diff
@@ -1,4 +1,29 @@
 import torch
+import torchtext
+from typing import List, Tuple
+
+
+def regex_tokenizer(line: str, patterns_list: List[Tuple[str, str]]) -> List[str]:
+    r"""Regex tokenizer for a string sentence.
+
+    Steps include
+    - lowercasing
+    - applying all regex replacements defines in patterns_list
+
+    Args:
+        line (str): a line of text to tokenize.
+        patterns_list (List[Tuple[str, str]]): a list of tuples (ordered pairs) which contain the regex pattern string as the first element and the replacement string as the second element.
+
+    Returns:
+        tokens (List[str]): a list of tokens after splitting on whitespace.
+    """
+
+    line = line.lower()
+    regex = torch.classes.torchtext.Regex('')
+    for pattern_re, replaced_str in patterns_list:
+        regex.UpdateRe(pattern_re)
```

Comment:
We could have a factory function that takes a list of regex and returns a list of Regex objects instead of this. Or maybe there's a way of writing a single regex from this list?

Reply:
Refer to above comment